w3af/plugins/grep/vulners_scanner.py

```diff
@@ -0,0 +1,219 @@
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with w3af; if not, write to the Free Software
+Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+
+"""
+
+# w3af object imports here
+import w3af.core.data.constants.severity as severity
+from w3af.core.controllers.plugins.grep_plugin import GrepPlugin
+from w3af.core.data.bloomfilter.scalable_bloom import ScalableBloomFilter
+from w3af.core.data.quick_match.multi_re import MultiRE
+from w3af.core.data.kb.vuln import Vuln
+import w3af.core.controllers.output_manager as om
+from w3af.core.data.options.opt_factory import opt_factory
+from w3af.core.data.options.option_types import STRING
+from w3af.core.data.options.option_list import OptionList
+
+# Additional modules imports
+import vulners
+import collections
+import re
+
+class vulners_scanner(GrepPlugin):
+    """
+    Find software vulnerabilities using Vulners.com API
+
+    Tests from the console mode:
+
+    With API key:
+
+    plugins grep vulners_scanner
+    plugins grep config vulners_scanner
+    set vulners_api_key YOUR_API_KEY_HERE
+    back
+    target set target VULNERABLE_WEBSITE
+    start
+
+
+    Without API key:
+
+    plugins grep vulners_scanner
+    target set target VULNERABLE_WEBSITE
+    start
+
+    :author: Vulners.com Team: Kir Ermakov (isox@vulners.com)
+    """
+
+    def __init__(self):
+        GrepPlugin.__init__(self)
+
+        # Vulners shared objects
+        self._vulners_api = None
+        self._vulners_api_key = None
+        self.rules_table = None
+        self.rules_updated = False
+
+        self._already_visited = ScalableBloomFilter()
+        self._vulnerability_cache = {}
+
+    def get_options(self):
+        """
+        :return: A list of option objects for this plugin.
+        """
+        ol = OptionList()
+
+        d = ('Vulners API key for extended scanning rate limits.'
+             'You can obtain one for free at https://vulners.com')
+        o = opt_factory('vulners_api_key', self._vulners_api_key, d, STRING)
+        ol.add(o)
+        return ol
+
+    def set_options(self, options_list):
+        """
+        This method sets all the options that are configured using the user
+        interface generated by the framework using the result of get_options().
+        :param options_list: A dictionary with the options for the plugin.
+        :return: No value is returned.
+        """
+        self._vulners_api_key = options_list['vulners_api_key'].get_value()
+
+
+    def update_vulners_rules(self):
+        # Get fresh rules from Vulners.
+        try:
+            self.rules_table = self.get_vulners_api().rules()
+            # Adapt it for MultiRe structure [(regex,alias)] removing regex duplicated
+            regexAliases = collections.defaultdict(list)
```

Comment:
Use _ instead of camel case.

regex_aliases
software_name

Reply:
Fixed.